File: BridgeCareApp/VuejsApp/src/shared/models/iAM/performance.ts
export interface PerformanceLibraryEquation {
    id: number | string;
    attribute: string;
    equationName: string;
    equation: string;
    criteria: string;
    shift: boolean;
    piecewise: boolean;
    isFunction: boolean;
}

export interface PerformanceLibrary {
    id: number | string;
    name: string;
    owner?: string;
    shared?: boolean;
    description: string;
    equations: PerformanceLibraryEquation[];
}

export const emptyEquation: PerformanceLibraryEquation = {
    id: 0,
    attribute: '',
    equationName: '',
    equation: '',
    criteria: '',
    shift: false,
    piecewise: false,
    isFunction: false,
};

export const emptyPerformanceLibrary: PerformanceLibrary = {
    id: 0,
    name: '',
    description: '',
    equations: []
};

File: BridgeCareApp/BridgeCare/EntityClasses/PERFORMANCE.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BridgeCare.EntityClasses
{
    using System;

    [Table("PERFORMANCE")]
    public class PERFORMANCE
    {
        [Key]
        public int PERFORMANCEID { get; set; }

        [ForeignKey("SIMULATION")]
        public int SIMULATIONID { get; set; }

        [ForeignKey("ATTRIBUTES_")]
        public string ATTRIBUTE_ { get; set; }

        public string EQUATIONNAME { get; set; }
        public string CRITERIA { get; set; }
        public string EQUATION { get; set; }
        public Nullable<bool> SHIFT { get; set; }
        public byte[] BINARY_EQUATION { get; set; }
        public byte[] BINARY_CRITERIA { get; set; }
        public Nullable<bool> PIECEWISE { get; set; }
        public Nullable<bool> ISFUNCTION { get; set; }

        public virtual Attributes ATTRIBUTES_ { get; set; }
        public virtual SIMULATION SIMULATION { get; set; }
    }
}